File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.java
class Solution {
    class Node{
        int ori;
        int change;
        int pos;

        public Node(int ori, int change, int pos) {
            this.ori = ori;
            this.change = change;
            this.pos = pos;
        }
    }
    
    public int[] sortJumbled(int[] mapping, int[] nums) {
        List<Node> list = new ArrayList<>();
        
        for (int i = 0; i<nums.length; i++){
            list.add(new Node(nums[i], modify(mapping, nums[i] + ""), i));
        }
        
        Collections.sort(list, (a, b) -> a.change == b.change ?  a.pos - b.pos : a.change - b.change);
        
        int[] res = new int[nums.length];
        for(int i=0; i<list.size(); i++){
            res[i] = list.get(i).ori;
        }
        
        return res;
    }
    
    private int modify(int[] mapping, String s) {
        char[] chars = s.toCharArray();
        for (int i = 0; i < chars.length; i++) {
            chars[i] = (char) (mapping[chars[i] - '0'] + '0');
        }

        return Integer.parseInt(new String(chars));
    }
}

File: 2191-sort-the-jumbled-numbers/2191-sort-the-jumbled-numbers.cpp
class Solution {
public:
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {   
        vector<pair<int,int>>vec;
        
        for(int i = 0;i<nums.size();++i){
            int num = nums[i];
          string number = to_string(num); 
            string formed = "";
          for(int j= 0;j<number.size();++j){
              formed+=(to_string(mapping[number[j]-'0']));
          }
          int value = stoi(formed);
          vec.push_back({value,i});
        }
        
        sort(vec.begin(),vec.end());
        vector<int>ans;
        for(auto pa:vec){
            int found = nums[pa.second];
            ans.push_back(found);
        }
        
        return ans;
    }
};
